services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -163,6 +185,8 @@ public void channelRead0(ChannelHandlerContext ctx, HttpRequest req) throws IOEx
       handleStreamReprocessingHybridStoreQuotaStatusLookup(ctx, helper);
     } else if (TYPE_STORE_STATE.equals(resourceType)) {
       handleStoreStateLookup(ctx, helper);
+    } else if (TYPE_REQUEST_TOPIC.equals(resourceType)) {
```

Comment:
time to enum and switchify these if-elses ?

Reply:
I'll try. I'm worried about the `diffCoverage` check 😅

Reply:
I attempted this and it's more than just `diffCoverage`. To do this right, there are changes that would affect the hot path too and I think we should address that in a separate PR. Let me know what you think

Reply:
Sure, its all over in read path currently.